src/main/java/com/minecolonies/coremod/util/constants/RenderUtils.java

```diff
@@ -0,0 +1,137 @@
+     * Render all waypoints.
+     * @param position the position of the build tool click.
+     * @param clientWorld the world.
+     * @param partialTicks the partial ticks
+     */
+    public static void renderWayPoints(final BlockPos position, final WorldClient clientWorld, final float partialTicks)
+    {
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(clientWorld, position);
+
+        if(colonyView == null)
+        {
+            return;
+        }
+
+        final Set<BlockPos> waypoints = colonyView.getWayPoints();
+        for (final BlockPos pos : waypoints)
+        {
+            final EntityEnderCrystal crystal = new EntityEnderCrystal(clientWorld);
+            crystal.setPosition(pos.getX() + HALF_BLOCK_OFFSET, pos.getY(), pos.getZ() + HALF_BLOCK_OFFSET);
+            crystal.setBeamTarget(pos.up(FIX_POINT_OFFSET));
+            crystal.setShowBottom(false);
+            crystal.innerRotation = 0;
+
+            Minecraft.getMinecraft().getRenderManager().renderEntityStatic(crystal, 0.0F, true);
+        }
+    }
+
+    /**
+     * Render the colony border.
+     * @param position the position of the structure.
+     * @param clientWorld the world.
+     * @param partialTicks the partial ticks.
+     * @param thePlayer the player clicking.
+     */
+    public static void renderColonyBorder(final BlockPos position, final WorldClient clientWorld, final float partialTicks, final EntityPlayerSP thePlayer)
+    {
+        if(colonyBorder.isEmpty())
+        {
+            calculateColonyBorder(clientWorld, thePlayer);
+        }
+
+        for (final BlockPos pos : colonyBorder)
+        {
+            final Block block = Blocks.DIAMOND_BLOCK;
+            final IBlockState iblockstate = block.getDefaultState();
+            final IBlockState iBlockExtendedState = block.getExtendedState(iblockstate, clientWorld, pos);
+            final IBakedModel ibakedmodel = Minecraft.getMinecraft().getBlockRendererDispatcher().getModelForState(iblockstate);
+            TileEntity tileentity = null;
+
+            final ModelHolder models = new ModelHolder(pos, iblockstate, iBlockExtendedState, tileentity, ibakedmodel);
+            Structure.getQuads(models, models.quads);
+            Settings.instance.getActiveStructure().renderGhost(clientWorld, models, thePlayer, partialTicks);
+        }
+    }
+
+    /**
+     * Calculate the colony border.
+     * @param theWorld in the world.
+     * @param thePlayer with the player.
+     */
+    private static void calculateColonyBorder(final WorldClient theWorld, final EntityPlayerSP thePlayer)
+    {
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(theWorld, thePlayer.getPosition());
+        if(colonyView == null)
+        {
+            return;
+        }
+        final BlockPos center = colonyView.getCenter();
+        final int radius = Configurations.workingRangeTownHall;
+
+        for ( int degrees = 0; degrees < 360; degrees += 1 )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 360 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)